### PR TITLE
fix: broken ci flow in Node 18

### LINKED ISSUE #902: CI failed on dashboards-search-relevance with TCP ECONNRESET
### What is the bug?
_A clear and concise description of the bug._

CI failed on dashboards-search-relevance with TCP ECONNRESET

### How can one reproduce the bug?
_Steps to reproduce the behavior._

This only happening on CI. Ex: https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/6280962123/job/17359747615?pr=267

```
Received 25036087 of 25036087 (100.0%), 57.7 MBs/sec
Cache Size: ~24 MB (25036087 B)
/usr/bin/tar -z -xf /home/runner/work/_temp/30538c1d-45c3-42d2-a001-0847c6b2bfe6/cache.tgz -P -C /home/runner/work/dashboards-search-relevance/dashboards-search-relevance
Cache restored successfully
Received 4[19](https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/6280962123/job/17359747615?pr=267#step:21:20)43040 of 368128980 (11.4%), 40.0 MBs/sec
Received 96468992 of 368128980 (26.2%), 46.0 MBs/sec
Received 197132288 of 368128980 (53.5%), 62.2 MBs/sec
Received 251658240 of 368128980 (68.4%), 59.6 MBs/sec
Received 339738624 of 368128980 (92.3%), 64.5 MBs/sec
Received 368128980 of 368128980 (100.0%), 60.7 MBs/sec
Cache Size: ~351 MB (368128980 B)
/usr/bin/tar -z -xf /home/runner/work/_temp/aeea8250-ab99-4718-a8c5-9[20](https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/6280962123/job/17359747615?pr=267#step:21:21)4d2ea5034/cache.tgz -P -C /home/runner/work/dashboards-search-relevance/dashboards-search-relevance
Cache restored successfully
/opt/hostedtoolcache/node/14.20.1/x64/bin/npm ci

> cypress@9.5.4 postinstall /home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test/node_modules/cypress
> node index.js --exec install

Note: Overriding Cypress cache directory to: ~/.cache/Cypress

      Previous installs of Cypress may not be found.


Cypress 9.5.4 is installed in /home/runner/.cache/Cypress/9.5.4


> opensearch-dashboards-functional-test@3.0.0 postinstall /home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test
> husky install

husky - Git hooks installed
added 329 packages in 10.792s
/opt/hostedtoolcache/node/14.20.1/x64/bin/npx cypress cache list
┌─────────┬───────────────────┐
│ version │ last used         │
├─────────┼───────────────────┤
│ 13.2.0  │ a few seconds ago │
├─────────┼───────────────────┤
│ 13.3.0  │ 4 days ago        │
├─────────┼───────────────────┤
│ 9.5.4   │ a few seconds ago │
└─────────┴───────────────────┘
Using custom test command: yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js
run tests command "yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js"
current working directory "/home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test"
/opt/hostedtoolcache/node/14.20.1/x64/bin/yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js
yarn run v1.22.10
$ /home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test/node_modules/.bin/cypress run --env SECURITY_ENABLED=false --browser chrome --spec 'cypress/integration/plugins/search-relevance-dashboards/*.js'
[5560:1003/163830.175363:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is swiftshader, ANGLE is 
Couldn't determine Mocha version

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        9.5.4                                                                          │
  │ Browser:        Chrome 117 (headless)                                                          │
  │ Node Version:   v14.20.1 (/opt/hostedtoolcache/node/14.20.1/x64/bin/node)                      │
  │ Specs:          1 found (plugins/search-relevance-dashboards/1_query_compare.spec.js)          │
  │ Searched:       cypress/integration/plugins/search-relevance-dashboards/*.js                   │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  plugins/search-relevance-dashboards/1_query_compare.spec.js                     (1 of 1)
Error: read ECONNRESET
    at TCP.onStreamRead (node:internal/stream_base_commons:[21](https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/6280962123/job/17359747615?pr=267#step:21:22)1:20)
 {
  errno: -104,
  code: 'ECONNRESET',
  syscall: 'read'
}
Error: read ECONNRESET
    at TCP.onStreamRead (node:internal/stream_base_commons:211:20)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: The process '/opt/hostedtoolcache/node/14.20.1/x[64](https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/6280962123/job/17359747615?pr=267#step:21:65)/bin/yarn' failed with exit code 1
```

### What is the expected behavior?
_A clear and concise description of what you expected to happen._

CI pass

```
> opensearch-dashboards-functional-test@3.0.0 postinstall /home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test
> husky install

husky - Git hooks installed
added 335 packages in 10.61s
/opt/hostedtoolcache/node/14.20.1/x64/bin/npx cypress cache list
┌─────────┬───────────────────┐
│ version │ last used         │
├─────────┼───────────────────┤
│ 13.0.0  │ a few seconds ago │
├─────────┼───────────────────┤
│ 13.1.0  │ 6 days ago        │
├─────────┼───────────────────┤
│ 9.5.4   │ a few seconds ago │
└─────────┴───────────────────┘
Using custom test command: yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js
run tests command "yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js"
current working directory "/home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test"
/opt/hostedtoolcache/node/14.20.1/x64/bin/yarn run cypress run --env SECURITY_ENABLED=false --browser chrome --spec cypress/integration/plugins/search-relevance-dashboards/*.js
yarn run v1.22.10
$ /home/runner/work/dashboards-search-relevance/dashboards-search-relevance/opensearch-dashboards-functional-test/node_modules/.bin/cypress run --env SECURITY_ENABLED=false --browser chrome --spec 'cypress/integration/plugins/search-relevance-dashboards/*.js'
[4722:0906/232051.235940:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is swiftshader, ANGLE is 
Couldn't determine Mocha version

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        9.5.4                                                                          │
  │ Browser:        Chrome 116 (headless)                                                          │
  │ Node Version:   v14.20.1 (/opt/hostedtoolcache/node/14.20.1/x64/bin/node)                      │
  │ Specs:          1 found (plugins/search-relevance-dashboards/1_query_compare.spec.js)          │
  │ Searched:       cypress/integration/plugins/search-relevance-dashboards/*.js                   │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  plugins/search-relevance-dashboards/1_query_compare.spec.js                     (1 of 1)
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
Couldn't determine Mocha version


  Compare queries
    ✓ Should get comparison results (46233ms)


  1 passing (51s)
```

### What is your host/environment?
_Operating system, version._

ubuntu-latest, 11

### Do you have any additional context?
_Add any other context about the problem._

A potential change to ignore such failures

https://github.com/hilleer/vscode-yaml-plus-json/pull/146/files#diff-2785793ab42443fdaafae000e605c53af6013c0ae7bc2aa3a490ba23f9238d69

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

For now all CI flows failed as OSD upgrade Node version to v18, which use ipv6 by default. Add --server.host="0.0.0.0" to enforce CI to setup OSD server in ipv4.

### Issues Resolved

#902 

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "VISBUILDER_ENABLED": true,
     "DATASOURCE_MANAGEMENT_ENABLED": false,
     "ML_COMMONS_DASHBOARDS_ENABLED": true,
-    "WAIT_FOR_LOADER_BUFFER_MS": 0
+    "WAIT_FOR_LOADER_BUFFER_MS": 0,
+    "NO_COMMAND_LOG": 1
```
Comment (ruanyl) [inline cypress.json:24]: Q: could you elaborate this change? seems not related to this PR

File: cypress/plugins/index.js [hunk 2]
```diff
@@ -24,4 +24,20 @@
 module.exports = (on, config) => {
   // `on` is used to hook into various events Cypress emits
   // `config` is the resolved Cypress config
+  on('before:browser:launch', (browser = {}, launchOptions) => {
```
Comment (ruanyl) [inline cypress/plugins/index.js:27]: The same here, could you elaborate a bit? seems not related to this PR

**Review by ruanyl — COMMENTED** [review]
Shall we make `--dns-result-order=ipv4first` the default option?

File: cypress.json (hunk 1 above)
Comment (SuZhou-Joe) [reply cypress.json:24]: On Chrome 117 the test will hang because of browser crash. According to the [doc](https://docs.cypress.io/guides/references/troubleshooting#Disable-the-Command-Log) here it seems the only option we can choose.

It is better to leave a comment here but it seems we can not write comment in json file.

File: cypress/plugins/index.js (hunk 2 above)
Comment (SuZhou-Joe) [reply cypress/plugins/index.js:32]: Sure, added comment above the code.

**Comment by SuZhou-Joe** [comment]
> Shall we make `--dns-result-order=ipv4first` the default option?

We should, and I moved the server="0.0.0.0" to NODE_OPTIONS env.

**Comment by ruanyl** [comment]
> > Shall we make `--dns-result-order=ipv4first` the default option?
> 
> We should, and I moved the server="0.0.0.0" to NODE_OPTIONS env.

I mean, shall we add it to `config/node.options` of OSD?
